src/tooltip/src/Tooltip.js

```diff
@@ -161,69 +111,100 @@ export default class Tooltip extends PureComponent {
     })
   }
 
-  isPopoverShown = () =>
-    // eslint-disable-next-line react/prop-types
-    this.props.popoverProps && this.props.popoverProps.isShown
+  // eslint-disable-next-line react/prop-types
+  const isPopoverShown = () => props.popoverProps && props.popoverProps.isShown
 
-  handleMouseEnterTarget = () => {
-    this.setState({
-      isShownByTarget: true
-    })
+  const handleMouseEnterTarget = () => {
+    setIsShownByTarget(true)
   }
 
-  handleMouseLeaveTarget = () => {
-    this.setState({
-      isShownByTarget: false,
-      willShow: false
-    })
-  }
+  let shown = (propIsShown || isShown || isShownByTarget) && !isPopoverShown()
 
-  render() {
-    const {
-      appearance,
-      isShown,
-      content,
-      position,
-      statelessProps = {}
-    } = this.props
-    const { isShown: stateIsShown, isShownByTarget } = this.state
-
-    let shown =
-      (isShown || stateIsShown || isShownByTarget) && !this.isPopoverShown()
-
-    // Tooltip was explicitly set to not be shown
-    if (isShown === false) {
-      shown = false
-    }
-
-    return (
-      <Positioner
-        target={({ getRef }) => {
-          return this.renderTarget({ getRef })
-        }}
-        isShown={shown}
-        position={position}
-        animationDuration={160}
-      >
-        {({ css, style, state, getRef }) => (
-          <TooltipStateless
-            id={this.state.id}
-            appearance={appearance}
-            ref={ref => getRef(ref)}
-            data-state={state}
-            style={style}
-            onMouseEnter={this.handleMouseEnterTarget}
-            onMouseLeave={this.handleMouseLeaveTarget}
-            {...statelessProps}
-            className={cx(
-              statelessProps.className,
-              css ? glamorCss(css).toString() : undefined
-            )}
-          >
-            {content}
-          </TooltipStateless>
-        )}
-      </Positioner>
-    )
+  // Tooltip was explicitly set to not be shown
+  if (propIsShown === false) {
+    shown = false
```

Comment:
Couldn't this be combined with the above to make things a bit clearer?

```suggestion
  let shown = typeof propIsShown === 'boolean'
    ? propIsShown
    : (isShown || isShownByTarget) && !isPopoverShown()
```